src/backends/linux_android_with_fallback.rs

```diff
@@ -54,7 +64,8 @@ fn init() -> NonNull<c_void> {
     res_ptr
 }
 
-// prevent inlining of the fallback implementation
+// Prevent inlining of the fallback implementation
+#[cold]
```

Comment:
Is this wise? I'm not sure what it does here but the fallback is not cold if it is used.

Reply:
Since we never inline the fallback function, this annotation just influences how compiler treats the `if fptr == NOT_AVAILABLE { .. } else { .. }` branch. Since we inline the `libc::getrandom` code, I think it makes sense to treat it as a more likely branch.

Reply:
Thinking about this a bit more, not inlining the fallback also seems like a mistake. The only thing that shouldn't be inlined is the *check* for whether or not fallback should be used. Otherwise we're paying a penalty for the fallback on every call.

I don't think we should hint the compiler about likelihood of fallback vs not.

Reply:
Maybe. I would be fine with removing `use_file_fallback` and using `use_file::fill_inner` directly in the if branch.

@josephlr WDYT?

Reply:
Maybe this discussion can happen in a later PR. I suggest removing the second commit for now and merging this to fix the breakage.